mailserver/mailserver_db_leveldb.go

```diff
@@ -68,7 +68,14 @@ func NewLevelDB(dataDir string) (*LevelDB, error) {
 		log.Info("database is corrupted trying to recover", "path", dataDir)
 		db, err = leveldb.RecoverFile(dataDir, nil)
 	}
-	return &LevelDB{ldb: db}, err
+	instance := LevelDB{ldb: db}
+
+	count, err := instance.envelopesCount()
+	if err == nil {
+		archivedEnvelopesCounter.Add(float64(count))
+	}
+
+	return &instance, err
```

Comment:
Do we want to surface this error? This means that an error when collecting metrics prevents from creating an instance of `LevelDB`. This is highly unlikely that `envelopesCount` returns an error and `LevelDB` instance is useful but it may happen.

So, if `envelopesCount` returns an error, you should either log it and ignore or return it but then return `nil` as the first value.

Currently, it's also inconsistent with `NewPostgresDB` constructor.

Reply:
Sure,I was thinking about ignoring it too. But if `count()` fails then the DB is clearly fucked, so it's better to fail the whole DB init.